Support insertMissing option in upsertGraph() for root models

lehni opened the pull request:
As discussed on [Gitter](https://gitter.im/Vincit/objection.js?at=59ec719532e080696e0eb50a), the `insertMissing` option in `upsertGraph()` currently does not work for root models.

This patch addresses it by leveraging the same mechanism already in place for child models.

I've started adding unit tests for `insertMissing: true` on root models, but am not sure if I'm doing it right, and also wanted to first check if this fix is the right way to go about the issue.

koskimas commented on tests/integration/upsertGraph.js:
```diff
@@ -969,7 +969,40 @@ module.exports = (session) => {
+      const upsert = {
+        // This doesn't exist.
+        id: 3,
+        model1Prop1: 'updated root 3',
```
Actually model with id 3 does exist. See the inserted object in `beforeEach`. Use some ridiculous number like 1234567 or the `NONEXISTENT_ID` constant instead.

koskimas commented:
This seems like a good approach if it works. The test fails currently. I added one comment that may be the cause.

lehni commented:
@koskimas I've addressed this now and also changed the test to run the upsert twice, first to insert the new root, then to upsert it. Let's see how this performs on Travis CI...

koskimas commented:
Awesome! Thank you!
